### ISSUE TITLE
No movement

### ISSUE THREAD

**Issue by everythingability** [drnoir/Papyrus3D-IF-Engine#1]
I installed locally and all seems to work except there's not WASD controls. (Mac M2, Ventura, Chrome)... I added them in the index.html file and now I can fly through walls.

Does Papyrus "do" collision btw... and how do I move if I don't use WASD/arrow keys?

The gun/blue box/key and door worked fine.

**Comment by drnoir** [comment]
Yes I took WASD out a while ago. You can add it back if you wanted it in, currently there is no physics system for collisions implemented but A-Frame does have A-Frame [aframe-physics-system](https://github.com/n5ro/aframe-physics-system) which uses Cannon.js which would probably work. The movement system I decided to go with a gaze based movement system for maximum multi platform support (VR  / AR) but it is arguably a trade off for sure. Might be worth me creating a branch for adding WASD and collisions in, it wouldn't be that complicated or possibly add a config  to switch between Gaze based movement and WASD. For collisions when the primitives are added one can add physics support by adding static body attribute. Would also need to add dynamic body to the player controller too for it to work correctly. From experience this can often be a pain with A-Frame at times but it's possible to add physics support

**Comment by drnoir** [comment]
https://github.com/drnoir/Papyrus3D-IF-Engine/tree/physics

Setup branch physics to see if can setup physics and collisions
Managed to get physics system setup with cannon.js but having issues with cam movement and char dynamic body with it so far. Will experiment more This week coming. Branch has debug mode on so static body is being applied to walls etc but not functioning completely with player cam

**Comment by everythingability** [comment]
I don't have easy access to VR headsets etc, so am a bit dumb in those areas, and haven't even played with gaze based movement... surely you still go through walls though?

I made a super-simple A-Frame maze-maker a while ago. 
- The p5js Map-maker is here: https://editor.p5js.org/tom.smith/sketches/RZ1RVSru5 <-- it generates a numbered matrix.
- The A-Frame project (that reads the data) is here: https://glitch.com/~a-frame-matrix-maze  <-- My version has walls that collide with the camera. See if you can find me dancing...

I found that to get  kinema.js to work (which handles simple collisions) I had to drop A-Frame down to version 0.8.2 or else it wouldn't work. I believe that the powers-that-be decided the way to implement this was using a mesh, so that it could be a floor that goes up stairs etc - whereas, I just wanted simple walls. So at that point, it kind of priced me out. I've no idea where to start with making meshes for A-Frame. I'm at the very basic level of everything 3D. 

Cheers, keep up the good work.

Tom
